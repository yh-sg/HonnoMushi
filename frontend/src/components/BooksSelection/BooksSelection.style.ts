import styled from "styled-components";

export const BooksSelectionStyle = styled.div`
	margin: 0 auto;
	width: 80vw;
`;
