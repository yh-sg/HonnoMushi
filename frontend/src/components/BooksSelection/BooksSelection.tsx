import React from "react";
import { Button, Row } from "react-bootstrap";
import { useHistory } from "react-router-dom";
import { BooksSelectionStyle } from "./BooksSelection.style";

const letters: string[] = "abcdefghijklmnopqrstuvwxyz".toUpperCase().split("");
const numbers: string[] = "0123456789".split("");

const BooksSelection = () => {
	let history = useHistory();
	return (
		<>
			<BooksSelectionStyle>
				<div className='row mt-3 mb-3'>
					<div>
						{letters.map((letter, index) => (
							<Button
								key={index}
								className='col-3 col-md-1 m-2 mb-3'
								variant='outline-success'
								onClick={() => history.push(`/books/${letter}`)}
								data-testid='letterButton'
							>
								{letter}
							</Button>
						))}
					</div>
				</div>

				<div className='row mb-3'>
					{numbers.map((number, index) => (
						<Button
							key={index}
							className='col m-2 p-2'
							variant='outline-primary'
							onClick={() => history.push(`/books/${number}`)}
							data-testid='numberButton'
						>
							{number}
						</Button>
					))}
				</div>
			</BooksSelectionStyle>
		</>
	);
};

export default BooksSelection;
