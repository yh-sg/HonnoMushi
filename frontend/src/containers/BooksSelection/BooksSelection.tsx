import React from "react";
import { Button, Row } from "react-bootstrap";
import { useHistory } from "react-router-dom";
import { BooksSelectionStyle } from "./BooksSelection.style";

const letters: string[] = "abcdefghijklmnopqrstuvwxyz".toUpperCase().split("");
const numbers: string[] = "0123456789".split("");

const BooksSelection = () => {
	let history = useHistory();
	return (
		<>
			<BooksSelectionStyle>
				<Row>
					{letters.map((letter, index) => (
						<Button
							key={index}
							className='col-2 m-4 p-2'
							variant='outline-success'
							onClick={() => history.push(`/books/${letter}`)}
						>
							{letter}
						</Button>
					))}
				</Row>
				<Row>
					{numbers.map((number, index) => (
						<Button
							key={index}
							className='col-2 m-4 p-2'
							variant='outline-primary'
							onClick={() => history.push(`/books/${number}`)}
						>
							{number}
						</Button>
					))}
				</Row>
			</BooksSelectionStyle>
		</>
	);
};

export default BooksSelection;
