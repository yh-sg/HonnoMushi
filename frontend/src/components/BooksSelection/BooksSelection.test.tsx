import { render, screen, cleanup } from "@testing-library/react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import BooksSelection from "./BooksSelection";

afterEach((): void => {
	cleanup();
});

describe("should render BooksSelection component correctly", () => {
	test("render button without crashing", () => {
		const div = document.createElement("div");
		ReactDOM.render(<BooksSelection></BooksSelection>, div);
	});

	test("should render alphabet buttons correctly", () => {
		const { getAllByTestId } = render(<BooksSelection />);
		const oneAlphabet = screen.getByText("Q");

		expect(getAllByTestId("letterButton")).toBeTruthy();
		expect(oneAlphabet).toBeInTheDocument();
	});

	test("should render number buttons correctly", () => {
		const { getAllByTestId } = render(<BooksSelection />);
		const oneNumber = screen.getByText("7");

		expect(getAllByTestId("numberButton")).toBeTruthy();
		expect(oneNumber).toBeInTheDocument();
	});

	// to update snapshot, press 'w' and 'u'
	test("should match snapshot", () => {
		const tree = renderer.create(<BooksSelection />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
