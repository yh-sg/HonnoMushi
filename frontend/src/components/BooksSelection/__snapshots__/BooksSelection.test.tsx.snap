// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BooksSelection component correctly should match snapshot 1`] = `
<div
  className="sc-bdnxRM cZnSuQ"
>
  <div
    className="row mt-3 mb-3"
  >
    <div>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        A
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        B
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        C
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        D
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        E
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        F
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        G
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        H
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        I
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        J
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        K
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        L
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        M
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        N
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        O
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        P
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Q
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        R
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        S
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        T
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        U
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        V
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        W
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        X
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Y
      </button>
      <button
        className="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
        data-testid="letterButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Z
      </button>
    </div>
  </div>
  <div
    className="row mb-3"
  >
    <button
      className="col m-2 p-2 btn btn-outline-primary"
      data-testid="numberButton"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="col m-2 p-2 btn btn-outline-primary"
      data-testid="numberButton"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="col m-2 p-2 btn btn-outline-primary"
      data-testid="numberButton"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="col m-2 p-2 btn btn-outline-primary"
      data-testid="numberButton"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="col m-2 p-2 btn btn-outline-primary"
      data-testid="numberButton"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="col m-2 p-2 btn btn-outline-primary"
      data-testid="numberButton"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="col m-2 p-2 btn btn-outline-primary"
      data-testid="numberButton"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="col m-2 p-2 btn btn-outline-primary"
      data-testid="numberButton"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="col m-2 p-2 btn btn-outline-primary"
      data-testid="numberButton"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="col m-2 p-2 btn btn-outline-primary"
      data-testid="numberButton"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      9
    </button>
  </div>
</div>
`;
